### PR TITLE
Added bazel BUILD file for test/core/memory_usage

### LINKED ISSUE #16852: missing bazel BUILD file: test/core/memory_usage/memory_usage_test.cc
https://github.com/grpc/grpc/blob/8ae3285323e126938bf46401ad09b9a17d3f1852/build.yaml#L3125

but no BUILD file here:
https://github.com/grpc/grpc/tree/bb44ca2c21a18fa48223228c73fa208e13eac44f/test/core/memory_usage

### PR DESCRIPTION
**Pull request by hcaseyal**
Fixes #16852

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,015,303      Total (=)      2,015,303

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,115,477      Total (<)     11,115,481

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

File: test/core/memory_usage/BUILD [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        "//:gpr",
+        "//:grpc",
+        "//test/core/util:gpr_test_util",
+        "//test/core/util:grpc_test_util",
+        "//test/core/end2end:ssl_test_data"
+    ],
+)
+
+grpc_cc_test(
+    name = "memory_profile_test",
```
Comment (jtattermusch) [inline test/core/memory_usage/BUILD]: actually, can we rename to "memory_usage_test"? I've recently renamed the test name in build.yam to match the name of the .cc file (see https://github.com/grpc/grpc/pull/16854/files)

Comment (hcaseyal) [reply test/core/memory_usage/BUILD]: Sure. Done.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,015,303      Total (=)      2,015,303

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,115,482      Total (>)     11,115,471

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Pull request merged**